src/core/api/link_api.cpp

```diff
@@ -493,7 +493,8 @@ otError otLinkCslSetChannel(otInstance *aInstance, uint8_t aChannel)
     otError   error    = OT_ERROR_NONE;
     Instance &instance = *static_cast<Instance *>(aInstance);
 
-    VerifyOrExit((Radio::kChannelMin <= aChannel) && (aChannel <= Radio::kChannelMax), error = OT_ERROR_INVALID_ARGS);
+    VerifyOrExit(aChannel == 0 || ((Radio::kChannelMin <= aChannel) && (aChannel <= Radio::kChannelMax)),
```

Comment:
update the doxygen for `otLinkCslSetChannel()` to allow value 0 to clear the csl channel?